pgmax/fg/graph.py

```diff
@@ -48,6 +48,7 @@ class FactorGraph:
         Mapping[Hashable, groups.VariableGroup],
         Sequence[groups.VariableGroup],
         groups.VariableGroup,
+        Mapping[int, groups.VariableGroup],
```

Comment:
This is very hacky....Could you investigate why this is happening and get a principled solution? I just tried and
```
from typing import Hashable
isinstance(2, Hashable)
```
gives me `True`. Also tests are still failing so this is not fixing the issue.

Reply:
There does not seem to be a principled solution to this problem. Atleast, I could not find one. So have included the hack for now. 
python/mypy#2412